DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -178,34 +177,34 @@ function Set-TargetResource
 
             if (-not (Test-RuleProperties -FirewallRule $firewallRule @PSBoundParameters))
             {
-                # Effectively renaming DisplayGroup to Group
-                if ($DisplayGroup) {
-                    $null = $PSBoundParameters.Add('Group', $DisplayGroup)
-                }
-
                 Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
                     $($LocalizedData.UpdatingExistingFirewallMessage) -f $Name
                     ) -join '')
 
-                # If the DisplayGroup is being changed then the rule
-                # Has to be removed and recreated because that is the
-                # Only way the DisplayGroup can be changed unfortunately.
-                # A change to the set-netfirewallrule has been requested:
-                # https://connect.microsoft.com/PowerShell/feedbackdetail/view/1970765/add-ability-to-change-firewall-displaygroup-in-set-netfirewallrule-cmdlet
+                # If the Group is being changed the the rule needs to be recreated
                 if ($PSBoundParameters.ContainsKey('Group') `
                     -and ($Group -ne $FirewallRule.Group)) {
-                    # Although it is possible to automatically remove and recreate
-                    # this rule, it introduces more problems than it solves.
-                    # So for now this will throw an error with the suggested solution.
-                    $errorId = 'CantChangeDisplayGroupError'
-                    $errorCategory = [System.Management.Automation.ErrorCategory]::InvalidOperation
-                    $errorMessage = $($LocalizedData.CantChangeDisplayGroupError) -f $Name
-                    $exception = New-Object -TypeName System.InvalidOperationException `
-                        -ArgumentList $errorMessage
-                    $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
-                        -ArgumentList $exception, $errorId, $errorCategory, $null
-
-                    $PSCmdlet.ThrowTerminatingError($errorRecord)
+
+                    Remove-NetFirewallRule -Name $Name
+
+                    # Merge the existing rule values into the PSBoundParameters
+                    # so that it can be splatted.
+
+                    # Loop through each possible property and if it is not passed as a parameter
+                    # then set the PSBoundParameter property to the exiting rule value.
+                    'DisplayName','Group','Enabled','Action','Profile','Direction', `
```

Comment:
Good idea! Done. I did look into auto pulling the properties list (it would need to get pulled from new-netfirewallrule), but because at the moment we're missing about 50% of the properties it didn't make sense. But once we've implemented all of them it will (although we'll still need to strip out ones implemented by the PSCmdLet base class and a few others).